torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -36,20 +36,14 @@ py::object PyRRef::toHere() {
   if (rref_->isOwner()) {
     return localValue();
   } else {
-    if (rref_->isPyObj()) {
-      // UserRRef<py::object>::toHere() calls python_rpc_handler which acquires
-      // GIL.
-      return jit::toPyObject(
```

Comment:
2 facts that help me understand `jit::toPyObject()` (which returns a `py::object`) requires GIL.

1. Constructing and destructing an py::object C++ object changes the Python object's ref count.
Ref: https://pybind11.readthedocs.io/en/stable/reference.html#_CPPv4N6object6objectERK6object

2. Changing Python objet's ref count requires acquiring GIL.
Ref: https://docs.python.org/3/c-api/init.html#thread-state-and-the-global-interpreter-lock

Combing them together, anywhere a py:object C++ object is present, the GIL should be acquired.